cockatrice/src/tab_deck_storage.cpp

```diff
@@ -112,6 +112,16 @@ TabDeckStorage::TabDeckStorage(TabSupervisor *_tabSupervisor, AbstractClient *_c
     setCentralWidget(mainWidget);
 }
 
+TabDeckStorage::~TabDeckStorage()
+{
+    emit deckStorageClosing(this);
```

Comment:
👎 

Let's do this reusably. Let's make a `ClosableTab` mixin class that adds an `emit tabClosing(this)` method. We don't need a differently-named copy of the same method across each tab class.